zcash_client_sqlite/src/wallet.rs

```diff
@@ -887,15 +1063,15 @@ fn subtree_scan_progress<P: consensus::Parameters>(
                 )
             })
             .transpose()?
-            // If none of the wallet's accounts have a recover-until height, then we can't
-            // (yet) distinguish general scanning from recovery, so treat the wallet as
-            // fully recovered.
-            .unwrap_or_else(|| Some(Ratio::new(1, 1)));
+            // If none of the wallet's accounts have a recover-until height, then there
+            // is no recovery phase for the wallet, and therefore the denominator in the
+            // resulting ratio (the number of notes in the recovery range) is zero.
+            .unwrap_or_else(|| Some(Ratio::new(0, 0)));
```

Comment:
This is fine for this API, but is an incompatible change for the SDK-level API. It's fine if we intend to explicitly handle 0:0 at the wallet level now; otherwise we should map that to 1:1 to emulate the old API.

Reply:
We _can_ cut an 0.13 as a hotfix if a breaking change is required. But I'm also inclined to _not_ make this breaking change, unless there's a bug-related reason for it.

Reply:
Currently the mapping to the existing API is done in zcash-light-client-ffi. That's where it makes most sense to do the mapping from 0:0 to 1:1, *after* adding the numerators and denominators, IMHO (i.e. in https://github.com/Electric-Coin-Company/zcash-light-client-ffi/pull/156).

Reply:
The scan progress denominator is guaranteed to always be nonzero, so adding the numerators and denominators always results in a valid fractional value.

Reply:
In pairing with @str4d, we determined that there may have already been uncommon but possible corner cases where the recovery ratio could have had a zero denominator: this would occur in circumstances where no notes for a given shielded protocol exist in blocks between the wallet's birthday height and the wallet recovery height. This is actually a relatively easily observable situation on testnet, since most of the time there's no shielded testnet activity. As such, `0:0` is not an API-breaking change; we just hadn't previously thought about or documented it adequately.